### PR TITLE
Update EditProject endpoint and tests

### PR DESCRIPTION
**Pull request by KostinP**

### CODE REVIEW

File: src/ProjectService.Validation/EditProjectValidator.cs [hunk 1]
```diff
@@ -1,25 +1,173 @@
...
+                            })
+                            .WithMessage("Project with this id is not exist.")
+                            .DependentRules(() =>
+                            {
+                                if ((bool)isActive)
+                                {
+                                    RuleFor(x => x.Patch.Operations)
+                                    .UniqueOperationWithAllowedOp(ClosedReasonPath, "add", "replace", "remove");
+
+                                    // If actual IsActive == true => reason for closing should not change
```
Comment (djino97) [inline src/ProjectService.Validation/EditProjectValidator.cs:137]: remove comments

File: src/ProjectService.Validation/EditProjectValidator.cs [hunk 2]
```diff
@@ -1,25 +1,173 @@
...
+                                        .Null()
+                                        .WithMessage($"The project is open, you cannot provide a {ClosedReasonPath} without closing the project.");
+                                    });
+                                }
+                                else
+                                {
+                                    RuleFor(x => x.Patch.Operations)
+                                    .UniqueOperationWithAllowedOp(ClosedReasonPath, "add", "replace");
+
+                                    // If actual IsActive == false => reason for closing can be replace
```
Comment (djino97) [inline src/ProjectService.Validation/EditProjectValidator.cs:150]: remove

File: src/ProjectService.Validation/EditProjectValidator.cs [hunk 3]
```diff
@@ -1,25 +1,173 @@
...
+                    });
+
+                    When(x => GetOperationByPath(x.Patch, IsActivePath) != null, () =>
+                    {
+                        RuleFor(x => x.Patch.Operations)
+                        .UniqueOperationWithAllowedOp(IsActivePath, "add", "replace");
+
+                        When(x => (bool)GetOperationByPath(x.Patch, IsActivePath).value, () =>
+                        {
+                            // If new IsActive == true => reason for the closure should be null
```
Comment (djino97) [inline src/ProjectService.Validation/EditProjectValidator.cs:88]: remove comments

File: src/ProjectService.Validation/EditProjectValidator.cs (hunk 1 above)
Comment (KostinP) [reply src/ProjectService.Validation/EditProjectValidator.cs:137]: Why?

File: src/ProjectService.Validation/EditProjectValidator.cs (hunk 2 above)
Comment (KostinP) [reply src/ProjectService.Validation/EditProjectValidator.cs:150]: Why?

File: src/ProjectService.Validation/EditProjectValidator.cs (hunk 3 above)
Comment (KostinP) [reply src/ProjectService.Validation/EditProjectValidator.cs:88]: Why?

**Review by MariyaSin — APPROVED** [review]

**Review by ParT1s — APPROVED** [review]

**Pull request merged**